File: leetcode/easy/121. Best Time to Buy and Sell Stock/solution.js
/** https://leetcode.com/problems/best-time-to-buy-and-sell-stock/
 * @param {number[]} prices
 * @return {number}
 */
function maxProfit(prices) {
    let min = prices[0], res = 0;
    for (let i = 1; i < prices.length; i += 1) {
        prices[i] < min 
            ? min = prices[i]
            : res = Math.max(res, prices[i] - min);
    }
    return res;
};

File: leetcode/easy/121. Best Time to Buy and Sell Stock/solution.cc
#include <vector>

using namespace std;

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minimum = prices[0], res = 0;
        for (int i = 0; i < prices.size(); ++i) {
            minimum > prices[i]
                ? minimum = prices[i]
                : res = max(res, prices[i] - minimum);
        }
        return res;
    }
};